components/gitpod-db/src/typeorm/auth-provider-entry-db-impl.ts

```diff
@@ -91,8 +91,10 @@ export class AuthProviderEntryDBImpl implements AuthProviderEntryDB {
         return query.getMany();
     }
 
-    protected oauthContentHash(oauth: AuthProviderEntry["oauth"]): string {
-        const result = createHash("sha256").update(JSON.stringify(oauth)).digest("hex");
+    protected oauthContentHash(entry: AuthProviderEntry): string {
+        const result = createHash("sha256")
+            .update(JSON.stringify({ oauth: entry.oauth, ownerId: entry.ownerId, status: entry.status }))
```

Comment:
Currently the code has no "update hash" path. If we change the meaning here we have to introduce one, or we get back into a state where we have not cashing at all!

Reply:
Don't get the comment.

HostContextProviderImpl is cashing the runtime handlers of auth providers. The original optimization was to reduce costs of decryption of db entries and instantiation of those runtime handlers. 

The problem here to be solved is, that the revision didn't consider rarely `ownerId`, so we overshoot and ignored necessary updates. So this basically adds the missing update, but it does not re-introduce re-reading of everything.

Reply:
> Don't get the comment.

Sorry for the confusion: did not read the code but answered from the top of my head - and that was outdated :see_no_evil: 

Hash gets updated [here](https://github.com/gitpod-io/gitpod/blob/45f13cf8e9214db664adc23a418e53d92364fbb7/components/server/src/auth/auth-provider-service.ts#L109) and [here](https://github.com/gitpod-io/gitpod/blob/45f13cf8e9214db664adc23a418e53d92364fbb7/components/server/src/auth/auth-provider-service.ts#L148).